src/app/nav/navigate-action.js

```diff
@@ -7,8 +13,16 @@ module.exports = function (actionContext, payload, done) {
   var route = actionContext.router.getRoute(payload.url);
   if (route) {
     actionContext.dispatch('navigate', route);
-    done();
+
+    // Execute page-specific initialize action:
+    var navigateStore = actionContext.getStore('navigate');
+    var currentPage = pageRegistry.getPage(navigateStore.getCurrentPage());
+    if (currentPage.initializeAction) {
+      actionContext.executeAction(currentPage.initializeAction, {}, done);
+    } else {
+      done();
+    }
   } else {
-    done({ failed: true });
+    done("No route matched.");
```

Comment:
Prefer consistency in quotes. I like (and think we've standardized on) single quotes.